### ISSUE TITLE
Improvement Request: AvoidModifyingObjectMapper - allow @PostConstruct

### ISSUE THREAD

**Issue by stokpop** [jborgers/PMD-jPinpoint-rules#116]
Modifying `ObjectMapper` in `@PostConstruct` should be allowed.

Assumption is that object mapper will only be used by other threads after Spring has created and called the `@PostConstuct` method. Also assume the `@PostConstruct` is not called directly from users code.

```java
@Component
public class MyComponent {

    private final ObjectMapper objectMapper = new ObjectMapper();

    @PostConstruct
    public void setUp() {
        objectMapper.setSerializationInclusion(JsonInclude.Include.NON_NULL); // good, should not be flagged
    }
```

note that this can also be fixed in current check using a static `ObjectMapper` and static init block:

```java
public class MyComponent {

    private final static ObjectMapper objectMapper;

    static {
        ObjectMapper newObjectMapper = new ObjectMapper();
        newObjectMapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
        objectMapper = newObjectMapper;
    }
```

(In some cases even better to create static thread safe `ObjectReader`s or `ObjectWriter`s.)
